src/mtcp/mtcp_restart.c

```diff
@@ -235,12 +238,18 @@ main(int argc, char *argv[], char **environ)
 #endif /* ifdef ENABLE_VDSO_CHECK */
 
   rinfo.fd = -1;
+  rinfo.mtcp_restart_pause = 0; /* false */
   rinfo.use_gdb = 0;
   rinfo.text_offset = -1;
   shift;
   while (argc > 0) {
+    // All parsing of flags with no arg must appear before parsing ckptImage
+    if (mtcp_strcmp(argv[0], "--mtcp-restart-pause") == 0) {
```

Comment:
@karya0, I think moving this to the end of the if-else would be a mistake.  It means that if we see `mtcp_restart <bunch of flags with args> --mtcp-restart-pause`, it currently reports `***MTCP Internal Error`.  In your revised ordering, it would report `***ERROR opening ckpt image (--mtcp-restart-pause); errno: No such file or directory` when it reaches lower in the code.  This type of thing can happen if DMTCP constructs an empty string for the checkpoint image file due to a logic error.